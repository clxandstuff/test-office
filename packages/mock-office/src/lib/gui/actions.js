export const START = 'app/START';
export const startAction = () => ({
  type: START
});
