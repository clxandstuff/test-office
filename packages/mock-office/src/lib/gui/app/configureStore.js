import { createStore, compose, applyMiddleware } from 'redux';
import { browserHistory } from 'react-router';
import { routerMiddleware as createRouterMiddleware } from 'react-router-redux';
import { createEpicMiddleware } from 'redux-observable';
import rootReducer from './rootReducer';
import rootEpic from '../epics';

const epicMiddleware = createEpicMiddleware(rootEpic);

const DEBUG = process.env.NODE_ENV !== 'production';

export default () => {
  const initialState = {};
  const routerMiddleware = createRouterMiddleware(browserHistory);
  const enhancer = compose(
    applyMiddleware(routerMiddleware, epicMiddleware),
    DEBUG && window.devToolsExtension ? window.devToolsExtension() : f => f
  );
  return createStore(rootReducer, initialState, enhancer);
};
