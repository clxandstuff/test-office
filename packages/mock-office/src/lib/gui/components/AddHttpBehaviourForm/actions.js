export const SUBMIT_SUCCEEDED = 'AddHttpBehaviourForm/SUBMIT_SUCCEEDED';
export const submitSucceededAction = (values, serverId) => ({
  type: SUBMIT_SUCCEEDED,
  values,
  serverId
});

export const SUBMIT_FAILED = 'AddHttpBehaviourForm/SUBMIT_FAILED';
export const submitFailedAction = reason => ({
  type: SUBMIT_FAILED,
  reason
});
