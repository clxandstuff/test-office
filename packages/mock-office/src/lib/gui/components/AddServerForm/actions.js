export const SUBMIT_SUCCEEDED = 'AddServerForm/SUBMIT_SUCCEEDED';
export const submitSucceededAction = values => ({
  type: SUBMIT_SUCCEEDED,
  values
});
