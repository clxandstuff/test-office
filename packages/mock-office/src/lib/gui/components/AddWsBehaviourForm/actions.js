export const SUBMIT_SUCCEEDED = 'AddWsBehaviourForm/SUBMIT_SUCCEEDED';
export const submitSucceededAction = (values, serverId) => ({
  type: SUBMIT_SUCCEEDED,
  values,
  serverId
});
