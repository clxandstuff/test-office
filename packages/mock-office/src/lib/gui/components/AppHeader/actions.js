export const EXPORT_BUTTON_CLICKED = 'AppHeader/EXPORT_BUTTON_CLICKED';
export const exportButtonClickedAction = () => ({
  type: EXPORT_BUTTON_CLICKED
});
