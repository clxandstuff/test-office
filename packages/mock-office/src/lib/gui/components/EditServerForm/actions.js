export const FORM_SUBMITTED = 'EditServerForm/FORM_SUBMITTED';
export const formSubmittedAction = (id, values) => ({
  type: FORM_SUBMITTED,
  id,
  values
});
