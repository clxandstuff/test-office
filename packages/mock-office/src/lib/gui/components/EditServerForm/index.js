import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import { Link } from 'react-router';
import serverIcon from '../../../../../assets/icons_gray_server.svg';
import { serverSelector } from '../../app/entities/index';
import { formSubmittedAction } from './actions';

export const EditServerForm = props => (
  <form className="form" action="#" onSubmit={props.handleSubmit}>
    <div className="form__header">
      <img className="form__header-icon" src={serverIcon} alt="" />
      Edit server
    </div>
    <div className="form-row">
      <div className="form__field">
        <label className="form-field__label" htmlFor="server-name">
          Name:
        </label>
        <Field
          component="input"
          className="form-field__input"
          name="name"
          type="text"
        />
      </div>
    </div>
    <div className="form-row">
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="server-port">
            Port:
          </label>
          <Field
            component="input"
            className="form-field__input"
            name="port"
            type="number"
            normalize={value => parseInt(value, 10)}
          />
        </div>
      </div>
    </div>
    <div className="form-row">
      <div className="form__col">
        <div className="form__field">
          <label className="form-field__label" htmlFor="server-port">
            Fallback url:
          </label>
          <Field
            component="input"
            className="form-field__input"
            name="fallbackUrl"
            type="text"
          />
        </div>
      </div>
    </div>
    <div className="form-row">
      <button className="button form__button" type="submit">
        Save
      </button>
      <Link to={`/server/${props.serverId}`} className="button form__button">
        Cancel
      </Link>
    </div>
  </form>
);

EditServerForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  serverId: PropTypes.string.isRequired
};

const mapDispatchToProps = (dispatch, ownProps) => ({
  onSubmit: values => dispatch(formSubmittedAction(ownProps.serverId, values))
});

const mapStateToProps = (state, ownProps) => {
  const server = serverSelector(state, ownProps.serverId);

  return {
    initialValues: {
      name: server.name,
      port: server.port,
      fallbackUrl: server.fallbackUrl
    }
  };
};

export const EditServerFormConnect = connect(
  mapStateToProps,
  mapDispatchToProps
)(
  reduxForm({
    form: 'editServer'
  })(EditServerForm)
);
