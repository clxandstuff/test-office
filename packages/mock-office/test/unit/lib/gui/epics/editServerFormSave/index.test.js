import { createEpicMiddleware } from 'redux-observable';
import configureStore from 'redux-mock-store';
import { formSubmittedAction } from '../../../../../../src/lib/gui/components/EditServerForm/actions';

afterEach(() => {
  jest.resetModules();
});

describe('editServerFormSave', () => {
  test('success actions', (done) => {
    jest.doMock('../../../../../../src/lib/gui/resources/mockOfficeService', () => ({
      editServer() {
        return Promise.resolve({});
      }
    }));
    const editServerFormSave = require('../../../../../../src/lib/gui/epics/editServerFormSave').default;
    const store = configureStore([createEpicMiddleware(editServerFormSave)])();
    store.dispatch(formSubmittedAction('id', { key: 'value' }));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  test('on InvalidRequestError actions', (done) => {
    const { InvalidRequestError } = require('../../../../../../src/lib/gui/packages/mock-office-service/lib/errors');
    jest.doMock('../../../../../../src/lib/gui/resources/mockOfficeService', () => ({
      editServer() {
        return Promise.reject(new InvalidRequestError());
      }
    }));
    const editServerFormSave = require('../../../../../../src/lib/gui/epics/editServerFormSave').default;
    const store = configureStore([createEpicMiddleware(editServerFormSave)])();
    store.dispatch(formSubmittedAction('id', { key: 'value' }));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  test('on UnsupportedStatusError actions', (done) => {
    const { UnsupportedStatusError } = require('../../../../../../src/lib/gui/packages/mock-office-service/lib/errors');
    jest.doMock('../../../../../../src/lib/gui/resources/mockOfficeService', () => ({
      editServer() {
        return Promise.reject(new UnsupportedStatusError());
      }
    }));
    const editServerFormSave = require('../../../../../../src/lib/gui/epics/editServerFormSave').default;
    const store = configureStore([createEpicMiddleware(editServerFormSave)])();
    store.dispatch(formSubmittedAction('id', { key: 'value' }));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  test('on ResourceNotFoundError actions', (done) => {
    const { ResourceNotFoundError } = require('../../../../../../src/lib/gui/packages/mock-office-service/lib/errors');
    jest.doMock('../../../../../../src/lib/gui/resources/mockOfficeService', () => ({
      editServer() {
        return Promise.reject(new ResourceNotFoundError());
      }
    }));
    const editServerFormSave = require('../../../../../../src/lib/gui/epics/editServerFormSave').default;
    const store = configureStore([createEpicMiddleware(editServerFormSave)])();
    store.dispatch(formSubmittedAction('id', { key: 'value' }));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });

  test('on ConnectionError actions', (done) => {
    const { ConnectionError } = require('../../../../../../src/lib/gui/packages/mock-office-service/lib/errors');
    jest.doMock('../../../../../../src/lib/gui/resources/mockOfficeService', () => ({
      editServer() {
        return Promise.reject(new ConnectionError());
      }
    }));
    const editServerFormSave = require('../../../../../../src/lib/gui/epics/editServerFormSave').default;
    const store = configureStore([createEpicMiddleware(editServerFormSave)])();
    store.dispatch(formSubmittedAction('id', { key: 'value' }));
    Promise.resolve().then(() => {
      expect(store.getActions()).toMatchSnapshot();
      done();
    });
  });
});
