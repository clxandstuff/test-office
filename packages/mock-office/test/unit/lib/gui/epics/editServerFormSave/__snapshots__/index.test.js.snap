// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editServerFormSave on ConnectionError actions 1`] = `
Array [
  Object {
    "id": "id",
    "type": "EditServerForm/FORM_SUBMITTED",
    "values": Object {
      "key": "value",
    },
  },
  Object {
    "reason": "Internet connection failed.",
    "type": "editServerFormSave/FAILED",
  },
]
`;

exports[`editServerFormSave on InvalidRequestError actions 1`] = `
Array [
  Object {
    "id": "id",
    "type": "EditServerForm/FORM_SUBMITTED",
    "values": Object {
      "key": "value",
    },
  },
  Object {
    "reason": "Something went wrong. Please try again or refresh page.",
    "type": "editServerFormSave/FAILED",
  },
]
`;

exports[`editServerFormSave on ResourceNotFoundError actions 1`] = `
Array [
  Object {
    "id": "id",
    "type": "EditServerForm/FORM_SUBMITTED",
    "values": Object {
      "key": "value",
    },
  },
  Object {
    "reason": "Something went wrong. Please try again or refresh page.",
    "type": "editServerFormSave/FAILED",
  },
]
`;

exports[`editServerFormSave on UnsupportedStatusError actions 1`] = `
Array [
  Object {
    "id": "id",
    "type": "EditServerForm/FORM_SUBMITTED",
    "values": Object {
      "key": "value",
    },
  },
  Object {
    "reason": "Something went wrong. Please try again or refresh page.",
    "type": "editServerFormSave/FAILED",
  },
]
`;

exports[`editServerFormSave success actions 1`] = `
Array [
  Object {
    "id": "id",
    "type": "EditServerForm/FORM_SUBMITTED",
    "values": Object {
      "key": "value",
    },
  },
  Object {
    "result": Object {},
    "type": "editServerFormSave/SUCCEEDED",
  },
  Object {
    "payload": Object {
      "args": Array [
        "/server/id",
      ],
      "method": "push",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;
